### PR TITLE
Basic otel trace support

### PR DESCRIPTION
**Pull request by sagikazarmark**
## Overview

This PR adds basic otel trace support for incoming requests. Integration for downstream requests (kafka, clickhouse) will be added in subsequent PRs.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -132,6 +137,31 @@ func main() {
...
+			os.Exit(1)
+		}
+
+		traceProviderOptions = append(traceProviderOptions, sdktrace.WithBatcher(exporter))
+	}
+
+	traceProvider := sdktrace.NewTracerProvider(traceProviderOptions...)
+	defer func() {
+		if err := traceProvider.Shutdown(ctx); err != nil {
+			logger.Error("shutting down trace provider: %v", err)
```
Comment (tothandras) [inline main.go:158]:
```suggestion
			logger.Error("shutting down trace provider", "error", err)
```

**Review by tothandras — CHANGES_REQUESTED** [review]
Should we make it optional? Seems like the telemetry address is required.

**Comment by sagikazarmark** [comment]
Telemetry address was required even before this change, but has a default value. We can think about making it optional later, but it doesn't cost anything.

**Comment by tothandras** [comment]
I see, I thought it was a push address.

**Review by tothandras — APPROVED** [review]

**Pull request merged**